core/src/main/java/ai/timefold/solver/core/impl/domain/valuerange/descriptor/FromListVarEntityPropertyValueRangeDescriptor.java

```diff
@@ -0,0 +1,73 @@
+package ai.timefold.solver.core.impl.domain.valuerange.descriptor;
+
+import ai.timefold.solver.core.api.domain.valuerange.ValueRange;
+import ai.timefold.solver.core.impl.domain.common.accessor.MemberAccessor;
+import ai.timefold.solver.core.impl.domain.valuerange.buildin.entity.FromEntityListValueRange;
+import ai.timefold.solver.core.impl.domain.valuerange.cache.CacheableValueRange;
+import ai.timefold.solver.core.impl.domain.variable.descriptor.GenuineVariableDescriptor;
+import ai.timefold.solver.core.impl.domain.variable.descriptor.ListVariableDescriptor;
+
+public class FromListVarEntityPropertyValueRangeDescriptor<Solution_>
+        extends FromEntityPropertyValueRangeDescriptor<Solution_> implements EntityIndependentValueRangeDescriptor<Solution_> {
+
+    private final ListVariableDescriptor<Solution_> listVariableDescriptor;
+
+    public FromListVarEntityPropertyValueRangeDescriptor(GenuineVariableDescriptor<Solution_> variableDescriptor,
+            boolean addNullInValueRange, MemberAccessor memberAccessor) {
+        super(variableDescriptor, addNullInValueRange, memberAccessor);
+        if (!(variableDescriptor instanceof ListVariableDescriptor<Solution_> descriptor)) {
+            throw new IllegalArgumentException(
+                    "The source of the value range must be instance of %s."
+                            .formatted(ListVariableDescriptor.class.getSimpleName()));
+        }
+        this.listVariableDescriptor = descriptor;
+    }
+
+    @Override
+    public boolean isEntityIndependent() {
+        return true;
+    }
```

Comment:
This makes no sense.
The descriptor allows extracting from both entity or solution. Therefore is can not be entity-independent. It also cannot be entity-dependent. Arguably, this distinction no longer makes any sense, and shouldn't be used.

Reply:
The logic has been updated to remove this method. However, we still need to determine whether a given value range is assigned to the solution or to the entity class. Therefore, a new method `ValueRangeDescriptor#canExtractValueRangeFromSolution` was added, and the related logic was updated accordingly.

The logic from `ValueRangeState` was also updated accordingly to handle value range extractions from solutions or entities.